packages/astro/src/transitions/router.ts

```diff
@@ -519,3 +525,53 @@ if (inBrowser) {
 		markScriptsExec();
 	}
 }
+
+// Keep all styles that are potentially created by client:only components
+// and required on the next page
+async function prepareForClientOnlyComponents(newDocument: Document, toLocation: URL) {
```

Comment:
have you manually run the build and verified that this code gets removed in prod? I think it will, since the only usage is wrapped in a conditional, but I don't understand how the tree shaking works well enough to know for sure.

Reply:
Sure, 6k unziped after tree shaking, 9k when not guarded by meta.env.DEV. Checked the content of the file manually to understand that it is really this tree shaking